### DISCUSSION TITLE
otlptracehttp.WithEndpoint is lacking the ability to provide path information.

### DISCUSSION THREAD

**Discussion by utezduyar** [open-telemetry/opentelemetry-go#3686] [General]
We have a domain abc.com/otel for the collector and SDK is missing the ability to use SDK's default trace path. The default path has to be explicitly specified. 

otlptracehttp.WithEndpoint is lacking the ability to put the path information to the endpoint. If it is used so, it includes the default trace path /v1/traces to the URL. If we specify WithURLPath("otelcol"), then the default trace path is not included by default. It needs to be WithURLPath("otelcol/v1/traces") like this. 

Wouldn't it make sense to export DefaultTracesPath or append the default trace path if the URLPath does not contain it.

**Comment by dmathieu** [comment]
I don't think this makes so much sense.

Appending the default trace path if it's not in URLPath means somebody needing a path which doesn't include this default value wouldn't be able to do so.
If I use `WithURLPath("/foobar")`, that means I want to send my data to `/foobar`, not `/foobar/v1/traces`.

As for exporting DefaultTracesPath, it wouldn't be too problematic to do so, but what would it bring?
If you use a custom url path, it means you want a specific value. Also, if a future version changes that default value, your endpoint's path won't change too. So you would start sending data to a missing path.

Hopefully this makes sense.

**Comment by utezduyar** [comment]
Thanks for your reply. Actually my concern is about a potential version change in the future. The one that is using the `WithURLPath` has to make a change in their code. The one that is not using will get the new path when they upgrade the opentelemetry go library. Your concern of sending data to a missing point is valid but I think it is already there for someone who is not using `WithURLPath`.

We were initially hosting our trace endpoint at `foo.com` and everything worked out of the box. I didn't need to know the existence of `v1/traces`. We have decided to move everything to a sub path as `foo.com/koo` and then I had to find out about `v1/traces` and append it with `WithURLPath`.

There is a difference in between 2 ways regarding how the default endpoint is gong to be changed. I prefer the upstream's new default and find out if it is working or not with my services.

**Reply by dmathieu** [reply]
The default value is set by the [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/1d014fb58f148d0dd5b8374cc8075d108c8cce02/specification/protocol/exporter.md#endpoint-urls-for-otlphttp).
That means in order to change it, the spec would have to change first. That specification is stable, and this is a breaking change. So that would mean a major release (v2).

I'm not saying it can't change. But that's very unlikely, and if it does, it will be because other quite large changes are introduced to the specification and therefore APIs.

**Reply by utezduyar** [reply]
I agree but I hope I was able to explain why exposing the DefaultTracesPath would make sense.
